### ISSUE TITLE
WMTS compatibility

### ISSUE THREAD

**Issue by mbarto** [geosolutions-it/csi-sira#780]
Check why this WMTS service is not working (works fine on latest MapStore DEV), we have an error when adding a layer to the map.
Is it related to the custom CRS?

Connected to #720

**Comment by dsuren1** [comment]
Hi @mbarto Kindly let me know the WMTS service tried

**Comment by mbarto** [comment]
Sorry, https://webgis.arpa.piemonte.it/ags/rest/services/topografia_dati_di_base/alluvione_20201003_S1_RGB/ImageServer/WMTS/1.0.0/WMTSCapabilities.xml

**Comment by dsuren1** [comment]
@mbarto Yes, you are right. This issue is because of the custom srs (EPSG:32632) which is not supported in mapstore (Supported ones are EPSG:900913 & EPSG:3857)

**Comment by mbarto** [comment]
Do you think it is difficult to implement support for other CRS in WMTS? I just need an effort evaluation (trivial, medium, difficult), then we can ask @tdipisa if we want to do it

**Comment by dsuren1** [comment]
The current WMTS capabilities URL shared doesn't have supporting TileMatrixSet for the custom CRS (EPSG:32632). Currently it only supports EPSG:3857.

**Comment by mbarto** [comment]
@dsuren1 Can you check this one? Seems to have issues too, in this case the EPSG:32632 grid should be supported:

http://geomap.reteunitaria.piemonte.it/mapproxy/service?

**Comment by dsuren1** [comment]
@mbarto For testing purpose, I bypassed the supported CRS validation in Mapstore to check if the layer can be added
I get an exception when making a call with _REQUEST=**DescribeDomains**_ 
"**unknown WMTS request type 'None'**" [service call ](http://geomap.reteunitaria.piemonte.it/mapproxy/service?service=WMTS&REQUEST=DescribeDomains&version=1.0.0&layer=pvto_igm25_1969_wmts&tileMatrixSet=EPSG%3A4326&expandLimit=10)
However the getCapabilities lists the only two operations . Could this be the reason for the calls failing with _REQUEST=**DescribeDomains**_ as the operation is not supported on the server? 
```
<ows:OperationsMetadata>
        <ows:Operation name="GetCapabilities">
        </ows:Operation>
        <ows:Operation name="GetTile">
        </ows:Operation>
    </ows:OperationsMetadata>
```
In this case, should we need to request based on tiles, then this request type is not available in Mapstore.
Kindly advise

**Comment by mbarto** [comment]
@dsuren1 I believe DescribeDomains is called to check if the layer supports the time dimension, but also if failing this should not prevent adding the layer to the map

**Comment by dsuren1** [comment]
@mbarto Got it. Seems like while calling the getTiles, the TileMatrixSet name is sent incorrect along with missing param 'tilematrix' is preventing from adding the layer successfully

**Comment by mbarto** [comment]
Let's see if you can fix this bug

**Comment by tdipisa** [comment]
@mbarto @dsuren1 yes, the DescribeDomain is performed to check if a time dimension is available. I can confirm what @mbarto said [here](https://github.com/geosolutions-it/csi-sira/issues/780#issuecomment-712694018).

**Comment by tdipisa** [comment]
@dsuren1 can you please better clarify if the problem for this issue is related to [this comment](https://github.com/geosolutions-it/csi-sira/issues/780#issuecomment-710100768) or [this one](https://github.com/geosolutions-it/csi-sira/issues/780#issuecomment-712712833)

**Comment by dsuren1** [comment]
Hi @tdipisa, this issue was closed earlier with this [comment](https://github.com/geosolutions-it/csi-sira/issues/780#issuecomment-710100768) unfortunately by the time I added the issue was close already. 
This [comment ](https://github.com/geosolutions-it/csi-sira/issues/780#issuecomment-712712833), is the actual issue and needs to be addressed in the Mapstore. I'm just investigating it a little and I should be able to give an estimate accordingly.

**Comment by dsuren1** [comment]
@tdipisa @mbarto The effort required to implement this is support is high based on my understanding.
- Support custom CRS WMTS layer add
- Investigate - With custom CRS, the WMTS layer calls stops with DescribeDomains (when fails) and doesn't trigger the operation of GetTile to fetch the tile information
- Formulate and add the missing params for the GetTile calls
- Analyze the regression impact

Kindly let me know your thoughts

**Comment by tdipisa** [comment]
@mbarto @dsuren1 I've involved @offtherailz for a double check on this. Thank you @dsuren1.

**Comment by offtherailz** [comment]
I report the result of my investigation.
First of all MapStore uses `ows:WGS84BoundingBox` to assign to the [WMTS source](https://openlayers.org/en/latest/apidoc/module-ol_tilegrid_WMTS-WMTSTileGrid.html ) `extent` and so prevent issues for tile loading outside the limit, or when origin/origins is not calculable.

This is the first issue, because for layers coming from the  bbox is wrong (this value is taken as `boundingBox` in creation of record and converted in the layer's `bbox`, then bbox is used in WMTSLayer to deternine the extent):
```
<ows:WGS84BoundingBox>
<ows:LowerCorner>-114.587420433 -2039.66243729</ows:LowerCorner>
<ows:UpperCorner>64.3425155404 1117.3210657</ows:UpperCorner>
</ows:WGS84BoundingBox>
```
Anyway this is an optional field that probably MapStore could have an option to ignore it.

Event removing this extent/bbox from the code (it is not mandatory), the tile requests are inconsinstent. 

This is one of them:
https://geomap.reteunitaria.piemonte.it/mapproxy/service?layer=regp_basecarto10bn_2015_wmts&style=&tilematrixset=grid_32632_nw&Service=WMTS&Request=GetTile&Version=1.0.0&Format=image%2Fpng&TileMatrix=14&TileCol=8039&TileRow=-8064

`TileRow=-8064` is negative, should be positive and with a different value. Not sure the reason of this problem, probably an issue determining origins or axis orientation for this tilematrix.

I think these are the things to pay attention to. Comparing with other WMTS in different SRS that are working.

**Comment by tdipisa** [comment]
@mbarto, I put it in blocked. This isn't realted to the MS project update but something to investigate MS side. I will open an issue MS side to schedule a fix for future MS releases.

**Comment by simboss** [comment]
@tdipisa please, like the othe issues here.
